clustered/common/src/main/java/org/ehcache/clustered/common/internal/store/WhiteListObjectInputStream.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.common.internal.store;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InvalidClassException;
+import java.io.ObjectInputStream;
+import java.io.ObjectStreamClass;
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+import static java.util.stream.Collectors.toList;
+
+public class WhiteListObjectInputStream extends ObjectInputStream {
+
+  private List<String> whiteListedClassNames;
+
+  public WhiteListObjectInputStream(InputStream in, Iterable<Class> whiteListedClasses) throws IOException {
+    super(in);
+    whiteListedClassNames = StreamSupport.stream(whiteListedClasses.spliterator(), false)
```

Comment:
Mapping to names seems a little hazardous.  A class name does not capture it's identity correctly.  It would seem to be safer to load the class (without initializing it) and do the proper check.  That would also allow us to switch to a set contains check which will be quicker.

Reply:
done

Reply:
The current version still uses class names to identify classes.  This seems wrong... we should be using Class instances which bind the name, the definition and the defining loader all together.

Reply:
as discussed , taking in Predicate<Class<?>> and classloader as input now